OPHD/Mine.cpp

```diff
@@ -2,9 +2,30 @@
 
 #include <iostream>
 
+#include <array>
+#include <map>
+
 using namespace NAS2D::Xml;
 
 
+/**
+ * Yield ore table
+ * 
+ * \note Follows the array layout conventions of the StorableResources class
+ * 
+ * [0] Common Metals
+ * [1] Common Minerals
+ * [2] Rare Metals
+ * [3] Rare Minerals
+ */
+static const std::map<MineProductionRate, std::array<int, 4>> YieldTable =
+{
+	{ MineProductionRate::Low, { 600, 500, 600, 500 }},
+	{ MineProductionRate::Medium, { 700, 550, 700, 550 }},
+	{ MineProductionRate::High, { 850, 600, 850, 600 }}
+};
```

Comment:
That's doable though even with using storable resources it would still need to be documented because it effectively acts as a wrapper around a std::array with a few related operations but doesn't enforce which array index correlates to which resource. That's up to the rest of the code. I opted to document it around the code where it's used. Granted, while this does open the possibility of failure to update documentation going forward I'm not too worried about it because I don't see the ordering changing in the future. At least not for this game.

Reply:
I would opt to document the order in one place, in the `StorableResources` struct. That way you don't need to update documentation in multiple places if things are ever changed or extended. It's super easy to miss comments when making updates, and the compiler doesn't in general check them, so this is important to consider. It should also be fairly easy to cross reference the documentation there, especially if the IDE supports ways to jump to the definition of a struct.